### PR TITLE
Fix to allow disabling P-RNG and SHA256 with CUSTOM_RAND_GENERATE_BLOCK

### PR DESCRIPTION
**Pull request by dgarske**
Example: ./configure --enable-cryptonly --disable-hashdrbg CFLAGS="-DNO_SHA256 -DCUSTOM_RAND_GENERATE_BLOCK" && make

### CODE REVIEW

File: wolfssl/wolfcrypt/random.h [hunk 1]
```diff
@@ -39,18 +39,22 @@
...
-    #ifdef NO_SHA256
-        #error "Hash DRBG requires SHA-256."
-    #endif /* NO_SHA256 */
 
-    #include <wolfssl/wolfcrypt/sha256.h>
-#else /* HAVE_HASHDRBG || NO_RC4 */
-    #include <wolfssl/wolfcrypt/arc4.h>
-#endif /* HAVE_HASHDRBG || NO_RC4 */
+#if defined(HAVE_HASHDRBG)
+    #ifdef NO_RC4
```
Comment (toddouska) [inline wolfssl/wolfcrypt/random.h]: This doesn't look right.  What if RC4 is on?  Then the SHA256 check doesn't happen.

Comment (dgarske) [inline wolfssl/wolfcrypt/random.h]: Thanks Todd for catching that. I had to review the RNG code again to figure out the different options. Corrected the bug and added some more inline commenting to help clarify things.

File: wolfssl/wolfcrypt/random.h [hunk 2]
```diff
@@ -39,18 +39,35 @@
...
 #if defined(WOLFSSL_FORCE_RC4_DRBG) && defined(NO_RC4)
     #error Cannot have WOLFSSL_FORCE_RC4_DRBG and NO_RC4 defined.
 #endif /* WOLFSSL_FORCE_RC4_DRBG && NO_RC4 */
-#if defined(HAVE_HASHDRBG) || defined(NO_RC4)
+
+
+/* RNG supports the following sources (in order):
+ * 1. CUSTOM_RAND_GENERATE_BLOCK: Defines name of function as RNG source and
+ *     bypasses the P-RNG.
+ * 2. HAVE_HASHDRBG && !SHA256 (default). Uses SHA256 based P-RNG
```
Comment (toddouska) [inline wolfssl/wolfcrypt/random.h]: The "!SHA256" in this comment seems incorrect.

**Comment by toddouska** [comment]
Thanks.

**Pull request merged**